internal/controlplane/grpc_accesslog.go

```diff
@@ -64,6 +69,16 @@ func populateLogEvent(
 	referer, _, _ := strings.Cut(entry.GetRequest().GetReferer(), "?")
 	path, query, _ := strings.Cut(entry.GetRequest().GetPath(), "?")
 
+	if !field.IsWellKnownField() && field.IsDynamicField() {
+		name, pathStr, _ := strings.Cut(string(field), "=")
+		name = strings.ToValidUTF8(strings.TrimSpace(name), "")
+		path, err := paths.Parse(entry.ProtoReflect().Descriptor(), pathStr)
```

Comment:
Is there a way to parse this path once and cache it? Or does the path depend on not only the descriptor but the contents of the log entry?

Reply:
(if the parsing is inexpensive then this may be premature optimization anyway)

Reply:
Yes we can cache these, the parsing step only operates on the descriptor, the dereference step operates on the actual message contents. The parsing is reasonably cheap but it would still make sense to cache it I think